internal/core/metadata/v2/application/device.go

```diff
@@ -130,3 +131,45 @@ func DeviceNameExists(name string, dic *di.Container) (exists bool, err errors.E
 	}
 	return exists, nil
 }
+
+// PatchDevice executes the PATCH operation with the device DTO to replace the old data
+func PatchDevice(dto dtos.UpdateDevice, ctx context.Context, dic *di.Container) errors.EdgeX {
+	dbClient := v2MetadataContainer.DBClientFrom(dic.Get)
+	lc := container.LoggingClientFrom(dic.Get)
+
+	var device models.Device
+	var edgeXerr errors.EdgeX
+	if dto.Id != nil {
+		device, edgeXerr = dbClient.DeviceById(*dto.Id)
+		if edgeXerr != nil {
+			return errors.NewCommonEdgeXWrapper(edgeXerr)
+		}
+	} else {
+		device, edgeXerr = dbClient.DeviceByName(*dto.Name)
```

Comment:
if get by name, check whether name is empty, or return 400 error

Reply:
Added validation and test case.